compiler/rustc_abi/src/layout.rs

```diff
@@ -954,9 +957,6 @@ fn univariant(
                         };
 
                         (
-                            // Place ZSTs first to avoid "interesting offsets", especially with only one
-                            // or two non-ZST fields. This helps Scalar/ScalarPair layouts.
-                            !f.0.is_zst(),
```

Comment:
This is the other place special-casing ZST that I removed

Reply:
This was present since https://github.com/rust-lang/rust/pull/45225 by @eddyb. Maybe at the time Scalar/ScalarPair computation still worked differently and the "interesting" offsets were an issue? It doesn't seem to be an issue any more today... :shrug:

Reply:
This was likely obviated by #73453